---
title: "Sashimi plot of mapped reads"
author: "SÃ¸ren Helweg Dam"
date: "2024-12-17"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


## BAM ----

Create a BAM list for the Sashimi plot

```{sh, eval=FALSE}
echo "Creating BAM list"
echo "ID  File  Patient Color
CRR025973	/home/projects/shd/relapseR/raw_data/bam/CRR025973Aligned.sortedByCoord.out.bam	Patient_A	1
CRR025971	/home/projects/shd/relapseR/raw_data/bam/CRR025971Aligned.sortedByCoord.out.bam	Patient_B	3
CRR025977	/home/projects/shd/relapseR/raw_data/bam/CRR025977Aligned.sortedByCoord.out.bam	Patient_C	5
CRR025975	/home/projects/shd/relapseR/raw_data/bam/CRR025975Aligned.sortedByCoord.out.bam	Patient_D	7
CRR025974	/home/projects/shd/relapseR/raw_data/bam/CRR025974Aligned.sortedByCoord.out.bam	Patient_A	2
CRR025972	/home/projects/shd/relapseR/raw_data/bam/CRR025972Aligned.sortedByCoord.out.bam	Patient_B	4
CRR025978	/home/projects/shd/relapseR/raw_data/bam/CRR025978Aligned.sortedByCoord.out.bam	Patient_C	6
CRR025976	/home/projects/shd/relapseR/raw_data/bam/CRR025976Aligned.sortedByCoord.out.bam	Patient_D	8" > /home/projects/shd/relapseR/raw_data/bam/bam_list.tsv
# rm /home/projects/shd/relapseR/raw_data/bam/bam_list.tsv
# vim /home/projects/shd/relapseR/raw_data/bam/bam_list.tsv
```


## Sashimi plot

```{sh, eval=FALSE}
echo "Creating sashimi plot"
cd /home/projects/shd/tools/
mkdir -p figs

echo "#4EEE94
#FF0000
#FFA500
#0000FF
#8FBC8F
#CD6090
#EECFA1
#9FB6CD" > ../relapseR/raw_data/bam/colors.tsv

python ggsashimi.py -b ../relapseR/raw_data/bam/bam_list.tsv -c 16:28931696-28933634 -o ../relapseR/figs/sashimi -g ../relapseR/raw_data/homo_sapiens/Homo_sapiens.GRCh38.107.gtf -L 3 -P ../relapseR/raw_data/bam/colors.tsv -C 4 --base-size 22


```


```{sh, eval=FALSE}
echo "ID\tFile\tPatient\tColor
19603_000001\t/home/projects/dp_immunoth/data/iCOPE/mapped/19603_000001.T.T02.S2_Aligned.sortedByCoord.out.bam\t19603_000001\t#4EEE94
20015_000000\t/home/projects/dp_immunoth/data/iCOPE/mapped/20015_000000.T.T02.S2_Aligned.sortedByCoord.out.bam\t20015_000000\t#4EEE94
21057_000001\t/home/projects/dp_immunoth/data/iCOPE/mapped/21057_000001.T.T02.S2_Aligned.sortedByCoord.out.bam\t21057_000001\t#4EEE94
21191_000000\t/home/projects/dp_immunoth/data/iCOPE/mapped/21191_000000.T.T02.S2_Aligned.sortedByCoord.out.bam\t21191_000000\t#4EEE94
21762_000000\t/home/projects/dp_immunoth/data/iCOPE/mapped/21762_000000.T.T02.S2_Aligned.sortedByCoord.out.bam\t21762_000000\t#4EEE94
21852_000000\t/home/projects/dp_immunoth/data/iCOPE/mapped/21852_000000.T.T02.S2_Aligned.sortedByCoord.out.bam\t21852_000000\t#4EEE94
22360_000000\t/home/projects/dp_immunoth/data/iCOPE/mapped/22360_000000.T.T02.S2_Aligned.sortedByCoord.out.bam\t22360_000000\t#4EEE94
23322_000000\t/home/projects/dp_immunoth/data/iCOPE/mapped/23322_000000.T.T02.S2_Aligned.sortedByCoord.out.bam\t23322_000000\t#4EEE94
23334_000000\t/home/projects/dp_immunoth/data/iCOPE/mapped/23334_000000.T.T02.S2_Aligned.sortedByCoord.out.bam\t23334_000000\t#4EEE94
23944_000000\t/home/projects/dp_immunoth/data/iCOPE/mapped/23944_000000.T.T02.S2_Aligned.sortedByCoord.out.bam\t23944_000000\t#4EEE94
24166_000000\t/home/projects/dp_immunoth/data/iCOPE/mapped/24166_000000.T.T02.S2_Aligned.sortedByCoord.out.bam\t24166_000000\t#4EEE94
24756_000000\t/home/projects/dp_immunoth/data/iCOPE/mapped/24756_000000.T.T02.S2_Aligned.sortedByCoord.out.bam\t24756_000000\t#4EEE94
27098_165905\t/home/projects/dp_immunoth/data/iCOPE/mapped/27098_165905.T.T02.S2_Aligned.sortedByCoord.out.bam\t27098_165905\t#4EEE94
27152_166111\t/home/projects/dp_immunoth/data/iCOPE/mapped/27152_166111.T.T02.S2_Aligned.sortedByCoord.out.bam\t27152_166111\t#4EEE94
27826_168775\t/home/projects/dp_immunoth/data/iCOPE/mapped/27826_168775.T.T02.S2_Aligned.sortedByCoord.out.bam\t27826_168775\t#4EEE94
7932_000000\t/home/projects/dp_immunoth/data/iCOPE/mapped/7932_000000.T.T02.S2_Aligned.sortedByCoord.out.bam\t7932_000000\t#4EEE94" > ../relapseR/data/icope_bam_list.tsv

python ggsashimi.py -b ../relapseR/data/icope_bam_list.tsv -c 16:28931696-28933634 -o ../relapseR/figs/sashimi_icope_noCD19 -g /home/projects/dp_immunoth/data/homo_sapiens/Homo_sapiens.GRCh38.107.gtf -L 3 -C 4 --base-size 22 --width 22
# Local
python3 ggsashimi.py -b ../../../data/iCOPE/mapped/7932_000000.T.T02.S2_Aligned.sortedByCoord.out.bam -c 16:28931696-28933634 -o ../relapseR/figs/sashimi_7932 -g ../../../data/homo_sapiens/Homo_sapiens.GRCh38.107.gtf -L 3 -C 4 --base-size 22 --width 22
```

21191_000000	../../../data/iCOPE/mapped/21191_000000.T.T02.S2_Aligned.sortedByCoord.out.bam	21191_000000### PSI

```{r psi}
# Load necessary libraries
library(ggplot2)
library(dplyr)
library(readr)

# Get a list of all files in the directory structure
# all_files <- list.files(path = "../data/zhang/outrigger", recursive = TRUE, full.names = TRUE, pattern = ".*outrigger_summary.csv")
all_files <- list.files(path = "~/mnt/cr2/people/s153398/relapseR/data/zhang/outrigger", recursive = TRUE, full.names = TRUE, pattern = ".*outrigger_summary.csv")

# Read, filter, and combine all matching files into a single data frame
exon2_data <- lapply(all_files, read_csv, show_col_types = FALSE, col_select = c(2:7), name_repair = "minimal") %>%
  bind_rows()
samples <- unique(exon2_data$sample_id)

delta_exon2 <- exon2_data |> 
  filter(grepl("16:28932089-28932910", event_id)) |> 
  mutate(psi = tidyr::replace_na(psi, 1))
  # select(sample_id, isoform1_junction13, psi)
delta_exon2 <- delta_exon2 |> 
  bind_rows(
    data.frame(
      sample_id = samples[!samples %in% delta_exon2$sample_id],
      psi = 1,
      isoform1_junction13 = 0
    )
  )

delta_exon2 |> 
  dplyr::filter(sample_id |> stringr::str_detect("CRR02597[3|1|7|5]")) |> 
  mutate(patient = factor(c("B", "A", "C", "D"), levels = c("A", "B", "C", "D"))) |> 
  select(sample_id, patient, isoform1_junction13) |> 
  ggplot() +
  aes(y = patient, x = isoform1_junction13) +
  geom_col()

# Plot the PSI distribution
ggplot(delta_exon2, aes(x = psi)) +
  geom_histogram(binwidth = 0.01, fill = "blue", color = "black", alpha = 0.7) +
  labs(title = "PSI of CD19exon2 - Zhang",
       x = "PSI",
       y = "Frequency") +
  ggpubr::theme_pubr()

# Plot Junction 1-3 distribution
ggplot(delta_exon2, aes(x = isoform1_junction13)) +
  geom_histogram(binwidth = 1, fill = "blue", color = "black", alpha = 0.7) +
  labs(title = "Junction 1-3 counts - Zhang",
       x = "Junction counts",
       y = "Frequency") +
  ggpubr::theme_pubr()
```


